iXBRLViewerPlugin/viewer/src/js/inspector.js

```diff
@@ -307,7 +307,7 @@ Inspector.prototype.update = function () {
         var a = new Accordian({
             onSelect: function (id) { 
                 inspector._currentFact = inspector._report.getFactById(id)
-                inspector._viewer.highlightFact(inspector._currentFact); 
+                inspector._viewer.highlightFact(inspector._currentFact.id); 
```

Comment:
If this line just used `id`, like so, could line 309 just be removed?

```
inspector._viewer.highlightFact(id); 
```

Reply:
I don't think so, at least not trivially.  `inspector.update()` on line 311 uses `inspector._currentFact` to update calculations, references, URL fragment, etc.

Reply:
👍

Reply:
Yes - assigning to this._currentFact changes the currently selected fact when the inspector is updated, but on reflection, this is messy.  I've refactored this in c55557c so we've now got Inspector.selectFact() which changes the current fact, and the current list of nested facts, and Inspector.switchFact() (called by selectFact()) which changes within the current list.

I've also refactored how the viewer interacts with the inspector as we had multiple routes to getting facts highlighted within the viewer.